admin/used_flusher.go

```diff
@@ -73,24 +100,48 @@ func (u *usedFlusher) flush() {
 	u.mu.Lock()
 	deployments := u.deployments
 	u.deployments = make(map[string]bool)
+
+	users := u.users
+	u.users = make(map[string]bool)
+
+	userTokens := u.userTokens
+	u.userTokens = make(map[string]bool)
+
+	serviceTokens := u.serviceTokens
+	u.serviceTokens = make(map[string]bool)
 	u.mu.Unlock()
 
-	if len(deployments) > 0 {
-		u.logger.Info("flushing used_on to db", zap.Int("deployments", len(deployments)))
+	// Helper function to perform the flushing of used_on to the database.
+	flushToDB := func(data map[string]bool, updateFn func(ctx context.Context, ids []string) error, logMsg string) {
```

Comment:
Instead of an inline function, move to a separate function like `func (u *usedFlusher) flushSingle(...` or something like that